docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/arithmetic/segment-max-16.rst

```diff
@@ -0,0 +1,102 @@
+SegmentMax
+===================
+
+
+.. meta::
+  :description: Learn about SegmentMax-16 - an arithmetic operation which computes the maximum values along segments of a tensor.
+
+**Versioned name**: *SegmentMax-16*
+
+**Category**: *Arithmetic*
+
+**Short description**: *SegmentMax-16* operation finds the maximum value in each specified segment of the ``input`` tensor.
+
+**Detailed description**
+
+For each index in ``segment_ids`` the operator gets values from ``data`` input tensor and calculates the maximum value for each segment.
+
+For example ``segments_ids`` with value ``[0,0,0,1,1,3,5,5]`` defines 4 non-empty segments. The other segments are empty. When coupled with a 1D data tensor ``data``, the segments are as follows:
+
+* Segment_0: ``[data[0], data[1], data[2]]``
+* Segment_1: ``[data[3], data[4]]``
+* Segment_2: ``[]``
+* Segment_3: ``[data[5]]``
+* Segment_4: ``[]``
+* Segment_5: ``[data[6], data[7]]``
+
+When there are no values in a segment, ``output[segment]`` is set to 0.
+
+In that case, the output would be ``[max(Segment_0), max(Segment_1), 0, max(Segment_3), 0, max(Segment_5)]``.
+
+**Attributes**:
+
+SegmentMax-16 has no attributes.
+
+**Inputs**
+
+* **1**: *data*
+
+  * **Description**: The numerical data on which SegmentMax operation will be performed. **Required.**
+  * **Range of values**: An ND tensor of type *T*.
+  * **Type**: *T*
+
+* **2**: *segment_ids*
+
+  * **Description**: Controls how the data is divided into segments. **Required.**
+  * **Range of values**: 1D tensor of non-negative, sorted integer numbers. Its size is equal to the size of the first dimension of the input tensor.
```

Comment:
I can see that unsorted `segment_ids` may lead to error or undefined behavior (implementation specific, depends on the hardware).
Should we specify a common behavior for OV op?
Can be clarified at the plugin implementation stage.

Reply:
I assumed that we'll verify the orderliness of `segment_ids` during shape inference and raise an error if needed. We can always remove that check from shape inference later and unify the behavior, but I suppose for now the op will be enabled in CPU with reusage of reference implementation.